tests/e2e/device/e2e_device_commands.c

```diff
@@ -584,13 +598,15 @@ static uint32_t prvE2ETestReportedPropertiesCommandExecute( E2E_TEST_COMMAND_HAN
     else if( AzureIoTHubClient_SendTelemetry( pxAzureIoTHubClient,
                                               xCMD->pulReceivedData,
                                               xCMD->ulReceivedDataLength,
-                                              NULL ) != eAzureIoTHubClientSuccess )
+                                              NULL, eAzureIoTHubMessageQoS1, &usTelemetryPacketID ) != eAzureIoTHubClientSuccess )

```

Comment:
how about creating send telemetry wrapper with timeout. In that function you can call for ProcessLoop till that timeout and check for packetId returned and received. To record the received packet have a global state that is shared between the callback and this wapper api?.

Using this way you do not need this complex, setting which might failed if process loop is not called and callback was not called.

Reply:
You're saying you don't want the ultracomplicated pointer pointer method???? :)

updated